x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValueSources.java

```diff
@@ -69,6 +69,18 @@ public static List<ValueSourceInfo> sources(
                 sources.add(new ValueSourceInfo(new NullValueSourceType(), new NullValueSource(), elementType, ctx.getIndexReader()));
                 continue; // the field does not exist in this context
             }
+            if (asUnsupportedSource) {
+                sources.add(
+                    new ValueSourceInfo(
+                        new UnsupportedValueSourceType(fieldType.typeName()),
+                        new UnsupportedValueSource(null),
```

Comment:
This is the actual fix where instead of using `null` as original source argument, the actual source was still in use.

Reply:
Thanks for explicitly calling this out `vs` -> `null`.  It would be easy to miss in review.